src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts

```diff
@@ -1162,3 +1178,136 @@ export class ObjectSettingWidget extends AbstractListSettingWidget<IObjectDataIt
+			this.model.select(null);
+			this.model.setEditOnly(true);
+			this.currentSettingKey = options.settingKey;
+		}
+
+		super.setValue(listData);
+	}
+
+	isItemNew(item: IBoolObjectDataItem): boolean {
+		return !item.key && !item.value;
+	}
+
+	protected getEmptyItem(): IBoolObjectDataItem {
+		return {
+			key: '',
+			value: false
+		};
+	}
+
+	protected getContainerClasses() {
+		return ['setting-list-object-widget'];
+	}
+
+	protected getActionsForItem(item: IBoolObjectDataItem, idx: number): IAction[] {
+		return [];
+	}
+
+	protected override isAddButtonVisible(): boolean {
+		return false;
+	}
+
+	protected override renderHeader() {
+		return undefined;
+	}
+
+	protected renderItem(item: IBoolObjectDataItem): HTMLElement {
+		// return empty object, since we always render in edit mode
+		const rowElement = $('.setting-list-row');
+		return rowElement;
+	}
+
+	protected renderEdit(item: IBoolObjectDataItem, idx: number): HTMLElement {
+		const rowElement = $('.setting-list-edit-row.setting-list-object-row.setting-item-bool');
+
+		const changedItem = { ...item };
+		const onValueChange = (newValue: boolean) => {
+			changedItem.value = newValue;
+			this.handleItemChange(item, changedItem, idx);
+		};
+		const { element, widget: checkbox } = this.renderEditWidget(changedItem.value, onValueChange);
+		rowElement.appendChild(element);
+
+		const valueElement = DOM.append(rowElement, $('.setting-list-object-value'));
+		const valueTitleElement = DOM.append(valueElement, $('.value-title'));
+		const { label } = settingKeyToDisplayFormat(changedItem.key);
+		valueTitleElement.textContent = label;
+		const valueDescriptionElement = DOM.append(valueElement, $('.value-description'));
+		if (item.description) {
+			valueDescriptionElement.textContent = item.description;
+		}
+		this._register(DOM.addDisposableListener(valueElement, DOM.EventType.MOUSE_DOWN, e => {
+			const targetElement = <HTMLElement>e.target;
+			if (targetElement.tagName.toLowerCase() !== 'a') {
+				checkbox.checked = !checkbox.checked;
+				onValueChange(checkbox.checked);
+			}
+			DOM.EventHelper.stop(e);
+		}));
+
+		return rowElement;
+	}
+
+	private renderEditWidget(
+		value: boolean,
+		onValueChange: (newValue: boolean) => void
+	) {
+		const checkbox = new Checkbox({
+			icon: Codicon.check,
+			actionClassName: 'setting-value-checkbox',
+			isChecked: value,
+			title: ''
+		});
+
+		this.listDisposables.add(checkbox);
+
+		const wrapper = $('.setting-list-object-input');
+		wrapper.classList.add('setting-list-object-input-key-checkbox');
+		checkbox.domNode.classList.add('setting-value-checkbox');
+		wrapper.appendChild(checkbox.domNode);
+
+		this._register(DOM.addDisposableListener(wrapper, DOM.EventType.MOUSE_DOWN, e => {
+			checkbox.checked = !checkbox.checked;
+			onValueChange(checkbox.checked);
+
+			// Without this line, the settings editor assumes
+			// we lost focus on this setting completely.
+			e.stopImmediatePropagation();
```

Comment:
What's going on here? And the checkbox should have its own listener

Reply:
This listener replaces the listener for the checkbox by giving it a similar behaviour as the `valueElement` listener. Currently, when focus is given to the checkbox, the entire setting seems to unfocus, the gear icon disappears, and the background darkens. Due to the settings re-rendering, focus is not preserved on the checkbox, either.
With this listener, focus is not preserved on the checkbox, but the other rendering side effects do not occur.

Reply:
> Currently, when focus is given to the checkbox, the entire setting seems to unfocus, the gear icon disappears, and the background darkens.

This seems fixable (we don't have to fix it in this PR) but let's look at it again later